### PR TITLE
[INLONG-5743][Agent] Support function read file data continuously through the monitoring file

### LINKED ISSUE #5743: [Feature][Agent] Read file data continuously through the monitoring file
### Description

Support function read file data continuously through the monitoring file

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by GanfengTan**
Support function read file data continuously through the monitoring file

Interval = 2s

- Fixes #5743 

### Motivation

No message

### Modifications

No message

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/TextFileReader.java [hunk 1]
```diff
@@ -46,6 +46,7 @@ public final class TextFileReader extends AbstractFileReader {
 
     public TextFileReader(FileReaderOperator fileReaderOperator) {
         super.fileReaderOperator = fileReaderOperator;
+        MonitorTextFile.getInstance().monitor(fileReaderOperator,this);
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/TextFileReader.java:49]: Add one blank after `,`, thanks.

File: inlong-agent/agent-plugins/src/test/resources/agent.properties [hunk 2]
```diff
@@ -23,5 +23,6 @@ job.thread.running.core=10
 agent.manager.vip.http.host=127.0.0.1
 agent.manager.vip.http.port=8083
 agent.fetcher.classname=org.apache.inlong.agent.plugin.fetcher.ManagerFetcher
-agent.manager.auth.secretId=
-agent.manager.auth.secretKey=
+agent.manager.auth.secretId=admin
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/test/resources/agent.properties:26]: Do those secret parameters need a default value?

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MonitorTextFile.class);
+    private static MonitorTextFile monitorTextFile;
+    private static Long INTERVAL = 2000L;
+    // monitor thread pool
+    private final ThreadPoolExecutor runningPool = new ThreadPoolExecutor(
+            0, Integer.MAX_VALUE,
+            60L, TimeUnit.SECONDS,
+            new SynchronousQueue<>(),
+            new AgentThreadFactory("monitor file"));
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java:44]: Suggested changing to `monitor-file`, as it was used for thread name, contains one blank is not a good practice.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.concurrent.TimeUnit;
+
+/**
+ * monitor files
+ */
+public final class MonitorTextFile {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MonitorTextFile.class);
+    private static MonitorTextFile monitorTextFile;
+    private static Long INTERVAL = 2000L;
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java:38]: Can this parameter be configured in the config file?

**Review by pocozh — COMMENTED** [review]
It seems that `canceling monitor mechanism` should be considered.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/TextFileReader.java (hunk 1 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/TextFileReader.java:49]: done

File: inlong-agent/agent-plugins/src/test/resources/agent.properties (hunk 2 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/test/resources/agent.properties:26]: UT property

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java (hunk 3 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java:44]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java (hunk 4 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            0, Integer.MAX_VALUE,
+            60L, TimeUnit.SECONDS,
+            new SynchronousQueue<>(),
+            new AgentThreadFactory("monitor-file"));
+
+    private MonitorTextFile() {
+
+    }
+
+    public static synchronized MonitorTextFile getInstance() {
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java:54]: For static members, initing in the static block is better.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        }
+        return monitorTextFile;
+    }
+
+    public void monitor(FileReaderOperator fileReaderOperator, TextFileReader textFileReader) {
+        MonitorEvent monitorEvent = new MonitorEvent(fileReaderOperator, textFileReader);
+        runningPool.execute(monitorEvent);
+    }
+
+    private class MonitorEvent implements Runnable {
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java:66]: Suggested changing to `MonitorEventRunnable`, and add Java doc, please.

**Review by pocozh — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java (hunk 5 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java:54]: Singleton

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java [hunk 7]
```diff
@@ -61,10 +61,10 @@ public class FileReaderOperator extends AbstractReader {
     public Stream<String> stream;
     public Map<String, String> metadata;
     public JobProfile jobConf;
-    private Iterator<String> iterator;
+    public Iterator<String> iterator;
+    public boolean finished = false;
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java:65]: Suggested adding `volatile` for `finished` field.

Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/FileReaderOperator.java:65]: done

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java (hunk 6 above)
Comment (GanfengTan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/file/MonitorTextFile.java]: done

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**